src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/consumer/impl/MessageConsumerImpl.java

```diff
@@ -46,7 +51,19 @@ void init() {
 
     @PreDestroy
     void close() {
-        consumer.close();
+        synchronized(lock) {
```

Comment:
if we just want to synchronize the consumer.close() call could we just have the method the way it was but declared as synchronized void close() with the @PreDestroy annotation. It looks a bit complicated otherwise.

Reply:
No I looked at that and it doe snot resolve the problem we still get that exception.